src/main/java/org/opensearch/performanceanalyzer/commons/collectors/ScheduledMetricCollectorsExecutor.java

```diff
@@ -120,7 +120,13 @@ public void run() {
                         }
                         metricsCollectors.put(
                                 collector, entry.getValue() + collector.getTimeInterval());
-                        if (!collector.inProgress()) {
+
+                        // exceptionCount is incremented in StatsCollector
+                        // and reset every time StatsCollector is run (i.e. every 60 sec)
+                        // since other collectors run every 5 sec, we can obtain an estimate
+                        // of the exceptions they've thrown in the last ~10 runs or so.
+                        int exceptionCount = StatsCollector.instance().exceptionCount(collector.getErrorMetric());
+                        if (!collector.inProgress() || exceptionCount > 5) {
```

Comment:
We can do `exceptionCount > 5` in the POC, but in the final implementation, this threshold should be configurable.